rollbar-android/src/main/java/com/rollbar/android/notifier/sender/ConnectionAwareSenderFailureStrategy.java

```diff
@@ -0,0 +1,188 @@
+  // confirm it via the ConnectivityManager, we will suspend sending occurrences for a second.
+  private static final long SUSPEND_TIME_UNKNOWN_ISSUE_MS = 1000;
+
+  ConnectionAwareSenderFailureStrategy(ConnectivityDetector detector,
+                                       NanoTimeProvider timeProvider) {
+    ObjectsUtils.requireNonNull(detector, "detector cannot be null");
+    ObjectsUtils.requireNonNull(timeProvider, "timeProvider cannot be null");
+
+    this.detector = detector;
+    this.timeProvider = timeProvider;
+    this.detector.setNetworkRestoredSignal(new Runnable() {
+      @Override
+      public void run() {
+        resume();
+      }
+    });
+
+    this.isSendingSuspended = false;
+  }
+
+  public ConnectionAwareSenderFailureStrategy(Context androidContext) {
+    this(new ConnectivityDetector(androidContext), new SystemNanoTimeProvider());
+  }
+
+  public void updateContext(Context androidContext) {
+    this.detector.updateContext(androidContext);
+  }
+
+  @Override
+  public PayloadAction getAction(Payload payload, Response response) {
+    NetworkStatus status = getNetworkStatus(response);
+    return getAction(status);
+  }
+
+  @Override
+  public PayloadAction getAction(Payload payload, Exception error) {
+    NetworkStatus status = getNetworkStatus(error);
+    return getAction(status);
+  }
+
+  @Override
+  public boolean isSendingSuspended() {
+    if (isSendingSuspended) {
+      if (timeProvider.nanoTime() >= resumeTime) {
+        resume();
+      }
+    }
+
+    return isSendingSuspended;
+  }
+
+  @Override
+  public void close() {
+    this.detector.close();
+  }
+
+  private PayloadAction getAction(NetworkStatus status) {
+    switch (status) {
+      case OK:
+        return PayloadAction.NONE;
+      case NO_NETWORK:
+        suspend(SUSPEND_TIME_NO_NETWORK_MS);
+        return PayloadAction.CAN_BE_RETRIED;
+      case UNKNOWN_NETWORK_ISSUE:
+        suspend(SUSPEND_TIME_UNKNOWN_ISSUE_MS);
+        return PayloadAction.CAN_BE_RETRIED;
+    }
+
+    return PayloadAction.NONE;
+  }
+
+  private void suspend(long suspendTimeMillis) {
+    isSendingSuspended = true;
+    resumeTime = timeProvider.nanoTime() + (suspendTimeMillis * 1000000L);
+  }
+
+  private void resume() {
+    isSendingSuspended = false;
+  }
+
+  private NetworkStatus getNetworkStatus(Response response) {
+    if (isSuccessfulResult(response.getResult())) {
+      return NetworkStatus.OK;
+    }
+    if (isSuspiciousHTTPStatus(response.getStatus())) {
+      return detector.isNetworkAvailable() ? NetworkStatus.OK : NetworkStatus.NO_NETWORK;
+    }
+
+    return NetworkStatus.OK;
+  }
+
+  private boolean isSuccessfulResult(Result result) {
+    if (result == null) {
+      return false;
+    }
+
+    return !result.isError();
+  }
+
+  private NetworkStatus getNetworkStatus(Throwable error) {
+    NetworkStatus status = NetworkStatus.OK;
+
+    if (error == null) {
+      return status;
+    }
```

Comment:
You provide the `status` object but don't update/use it later in the function.

```suggestion
    if (error == null) {
      return NetworkStatus.OK;
    }
```

or update the `status` across the function and return it at the end.